### ISSUE TITLE
Passing the field: `min_instances` does not get applied to the cloud function

### ISSUE THREAD

**Issue by jlazar** [google-github-actions/deploy-cloud-functions#259]
### TL;DR

Passing a `min_instances` of 1 does not modify the actual min instances of the cloud function. 

Below step will not work:
```
    - id: 'deploy'
      uses: 'google-github-actions/deploy-cloud-functions@v0'
      with:
        name: 'my-function'
        runtime: 'nodejs12'
        min_instances: 1
```
It seems the root of this issue is that the Cloud Functions API does not have a minInstances defined in its spec: https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions#CloudFunction

### Expected behavior

Correctly set the min_instances of the cloud function

### Observed behavior

min_instances for the cloud function was set to 0

### Action YAML

```yaml
name: "Cloud Functions Deploy"

on:
  push:
    branches: [develop]
    paths:
      - functions/**

jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        function:
          - test1

        include:
          - function: test1
            name: test1
            entry_point: test1
            min_instances: 1

    steps:
      - uses: actions/checkout@v2

      - name: install node v14
        uses: actions/setup-node@v1
        with:
          node-version: 14

      - name: yarn install
        run: yarn install

      - name: Typescript compiler
        run: yarn functions build

      - name: Setup GCP Service Account
        uses: google-github-actions/auth@v0.4.1
        with:
          credentials_json: ${{secrets.GOOGLE_CLOUD_GITHUB_SERVICE_ACCOUNT}}

      - uses: google-github-actions/deploy-cloud-functions@main
        with:
          name: ${{matrix.name}}
          event_trigger_type: ${{matrix.event_trigger_type}}
          event_trigger_resource: ${{matrix.event_trigger_resource}}
          entry_point: ${{matrix.entry_point}}
          runtime: nodejs14
          region: us-east4
          min_instances: ${{matrix.min_instances}}
```

**Comment by sethvargo** [comment]
It's not in the auto-generated docs, but it's present in the [discovery doc](https://cloudfunctions.googleapis.com/$discovery/rest?version=v1). 

What are you seeing as the value for `minInstances` when you query the function?

**Comment by jlazar** [comment]
@sethvargo thanks, when I use `gcloud functions describe [my_function]` I do not get a `minInstances` key or value.  When I re-deploy the same function via google console with minInstances=1 and then run `gcloud functions describe [my_function]` I do see `minInstances=1`.

**Comment by sethvargo** [comment]
@jlazar the default gcloud output only includes a subset of fields. Try with `--format=yaml` or `--format=json` to see all fields.

**Comment by jlazar** [comment]
@sethvargo providing my stripped down output below. After a few tests, I think this could be only an issue with the `PATCH` flow. If i delete my function and re-create with the `minInstances` params set; it will create properly. But if I omit `minInstances` on the initial create and then update to add, no luck. I believe it may need to be added to the `updateMasks` array here: https://github.com/google-github-actions/deploy-cloud-functions/blob/main/src/client.ts#L405

I am also seeing this same flow with the param: `secretEnvironmentVariables`. Create works, update (patch) does not

Cleaned output from `gcloud functions describe foo --format=json`
```
{
  "availableMemoryMb": 256,
  "buildId": "foo",
  "buildName": "foo",
  "entryPoint": "foo",
  "environmentVariables": {
    "ENV": "stg"
  },
  "httpsTrigger": {
    "securityLevel": "foo",
    "url": "foo"
  },
  "ingressSettings": "ALLOW_ALL",
  "name": "foo",
  "runtime": "nodejs14",
  "secretEnvironmentVariables": [...]
  "serviceAccountEmail": "foo",
  "sourceUploadUrl": "foo",
  "status": "ACTIVE",
  "timeout": "60s",
  "updateTime": "foo",
  "versionId": "2"
}
```

**Comment by sethvargo** [comment]
Yea, updateMasks is the issue. You're totally right.

**Comment by sethvargo** [comment]
Fixed in #260
